src/qobj.jl

```diff
@@ -24,92 +26,60 @@ end
 function generate_inst(qc_simpl::ChainBlock)
     inst = []
     for block in subblocks(qc_simpl)
-        if block isa Union{PutBlock{N, M, ChainBlock{M}}, ChainBlock{O}} where {N, M, O}
-            push!(inst, generate_inst(block)...)
+        i = generate_inst(block)
+        if block isa ChainBlock
+            if i isa Array{Array}                             # for nested chains
+                push!(inst, Iterators.flatten(i)...)  #generalize this to loop till we get a Array{Dict} 
+            else
+                append!(inst, i)
+            end
         else
-            push!(inst, generate_inst(block))
+            push!(inst, i)
         end
     end
     return inst
 end
 
-function generate_inst(blk::PutBlock)
-    gate = blk.content
-    if gate isa HGate
-        nm = "h"
-    elseif gate isa I2Gate
-        nm = "id"
-    elseif gate isa TGate
-        nm = "t"
-    elseif gate isa SWAPGate
-        nm = "swap"
-    elseif gate isa Measure
-        nm = "measure"
-    end
-
-    if nm == "measure"
-        return Dict("name"=>"$(nm)", "qubits"=>[blk.locs...], "memory"=>[0])# memory:  List of memory slots in which to store the measurement results (mustbe the same length as qubits).  
-        #todo: generalize "memory" for all cases               
-    else    
-        return Dict("name"=>"$(nm)", "qubits"=>[blk.locs...]) 
-    end
+function generate_inst(blk::PutBlock{N,M}) where {N,M}
+	locs = [blk.locs...]
+	generate_instv2(blk.content, locs)
 end
 
-function generate_inst(blk::PutBlock{N, M, ChainBlock{M}}) where {N, M}
-    data = []
-    for gate in blk.content
-        super_blk = gate.block
-        if super_blk isa ZGate
-            nm = "rz"
-        elseif super_blk isa YGate
-            nm = "ry"
-        elseif super_blk isa XGate
-            nm = "rx"
-        end
-        inst = Dict("name"=>"$(nm)", "qubits"=>[blk.locs...], "params"=>[gate.theta])
-        push!(data, inst)
-    end
-    return Tuple(data)
+function generate_inst(blk::ControlBlock{N,GT,C}) where {N,GT,C}
+	generate_instv2(blk.content, blk.locs, blk.ctrl_locs)
```

Comment:
Is this a legacy code?

Reply:
No. It handles the dispatch for control block.

Reply:
I mean the `v2`.